invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/components/SharedOrMineFilter.js

```diff
@@ -0,0 +1,110 @@
+// This file is part of Invenio
+// Copyright (C) 2023 CERN.
+//
+// Invenio is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import { i18next } from "@translations/invenio_app_rdm/i18next";
+import PropTypes from "prop-types";
+import React, { Component } from "react";
+import { withState } from "react-searchkit";
+import { Dropdown, Button } from "semantic-ui-react";
+
+class SharedOrMineFilterComponent extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      sharedWithMe: false,
+    };
+  }
+
+  componentDidMount() {
+    const { currentQueryState } = this.props;
+    const userSelectionFilters = currentQueryState.filters;
+    const sharedWithMe = userSelectionFilters.find((obj) =>
+      obj.includes("shared_with_me")
+    );
+    if (sharedWithMe) {
+      // eslint-disable-next-line react/no-did-mount-set-state
```

Comment:
could this part be moved to the constructor? you only need access to props, is it any way dependant on the fact if the component mounted or not?
The reason why eslint does not like to set state inside the DidMount: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-did-mount-set-state.md